contrib/valgrind/valgrind_vma.supp

```diff
@@ -0,0 +1,161 @@
+
+{
+   Conditional_on_uninitialised_value
+   Memcheck:Cond
+   fun:_ZN11buffer_pool11get_buffersEmj
+}
+{
+   Points_to_uninitialised_bytes
+   Memcheck:Param
+   epoll_ctl(event)
+   fun:epoll_ctl
+}
+{
+   Conditional_on_uninitialised_value
+   Memcheck:Cond
+   fun:is_valid_fd
+   ...
+   fun:ring_eth
+}
+{
+   Conditional_on_uninitialised_value
+   Memcheck:Cond
+   fun:is_valid_fd
+   ...
+   fun:~ring_eth
+}
+{
+   Conditional_on_uninitialised_value
+   Memcheck:Cond
+   fun:is_valid_fd
+   fun:get<epfd_info>
+   fun:get_epfd
+}
+{
+   Conditional_on_uninitialised_value
+   Memcheck:Cond
+   fun:is_valid_fd
+   fun:get<socket_fd_api>
+   fun:get_sockfd
+}
+{
+   Conditional_on_uninitialised_value
+   Memcheck:Cond
+   fun:is_valid_fd
+   fun:get<cq_channel_info>
+   fun:get_cq_channel_fd
+}
```

Comment:
Per my opinion the latest leak should be fixed too and removed from suppression file. Probably in destructor we can change state machine to desired state and call cleanup function.